File: python/samples/01-SimpleChat/__main__.py
import os
import sys
import asyncio

from semantic_kernel.utils.settings import azure_openai_settings_from_dot_env_as_dict
# to allow the strange structure and the import of the new pieces
sys.path.append(os.getcwd())
from python.src.kernel import newKernel as Kernel
from python.src.sk_function import SKFunction
from python.src.azure_chat_completion import AzureChatCompletion

async def runner():
    # create services and chat
    gpt35turbo = AzureChatCompletion(
        **azure_openai_settings_from_dot_env_as_dict(include_api_version=True),
    )
    chat_function = SKFunction.from_yaml(
        os.getcwd()
        + "/python/samples/01-SimpleChat/plugins/ChatPlugin/SimpleChat.prompt.yaml"
    )

    # create kernel
    kernel = Kernel(ai_services=[gpt35turbo])

    # create chat_history
    chat_history = gpt35turbo.create_new_chat()
    chat_history.add_system_message("Hello! I am a robot.")
    chat_history.add_user_message("Hello! I am a human.")
    # loop with input
    while True:
        user_input = input("User:> ")
        if user_input == "exit":
            break
        chat_history.add_user_message(user_input)

        # get response
        response = await kernel.run_async(
            chat_function,
            variables={"messages": chat_history},
            service=gpt35turbo
        )

        # print response
        print(f"Assistant:> {response['result']}")
        chat_history.add_openai_response(response['response_object'])

def __main__():
    asyncio.run(runner())

if __name__ == "__main__":
    __main__()

File: java/samples/01-SimpleChat/src/main/java/Main.java
import java.nio.file.Path;

import com.azure.ai.openai.OpenAIAsyncClient;
import com.azure.ai.openai.OpenAIClientBuilder;
import com.azure.core.credential.KeyCredential;
import com.microsoft.semantickernel.Kernel;
import com.microsoft.semantickernel.KernelResult;
import com.microsoft.semantickernel.SKBuilders;
import com.microsoft.semantickernel.chatcompletion.ChatCompletion;
import com.microsoft.semantickernel.chatcompletion.ChatHistory;
import com.microsoft.semantickernel.exceptions.ConfigurationException;
import com.microsoft.semantickernel.orchestration.ContextVariables;
import com.microsoft.semantickernel.orchestration.SKFunction;
import com.microsoft.semantickernel.v1.semanticfunctions.SemanticFunction;
import com.microsoft.semantickernel.v1.templateengine.HandlebarsPromptTemplateEngine;

public class Main {
    
    final static String GPT_35_DEPLOYMENT_NAME = System.getenv("GPT_35_DEPLOYMENT_NAME");
    final static String GPT_4_DEPLOYMENT_NAME = System.getenv("GPT_4_DEPLOYMENT_NAME");
    final static String AZURE_OPENAI_ENDPOINT = System.getenv("AZURE_OPENAI_ENDPOINT");
    final static String AZURE_OPENAI_API_KEY = System.getenv("AZURE_OPENAI_API_KEY");
    final static String CURRENT_DIRECTORY = System.getProperty("user.dir");
    
    
    public static void main(String[] args) throws ConfigurationException {

        OpenAIAsyncClient client = new OpenAIClientBuilder()
            .credential(new KeyCredential(AZURE_OPENAI_API_KEY))
            .endpoint(AZURE_OPENAI_ENDPOINT)
            .buildAsyncClient();
        // Initialize the required functions and services for the kernel
        Path yamlPath = Path.of(CURRENT_DIRECTORY + "/Plugins/ChatPlugin/PersonaChat.prompt.yaml");
        SKFunction chatFunction = SemanticFunction.fromYaml(yamlPath);

        ChatCompletion<ChatHistory> gpt35Turbo = ChatCompletion.builder()
            .withOpenAIClient(client)
            .withModelId(GPT_35_DEPLOYMENT_NAME)
            .build();

        ChatCompletion<ChatHistory> gpt4 = ChatCompletion.builder()
                .withOpenAIClient(client)
                .withModelId(GPT_4_DEPLOYMENT_NAME)
                .build();


        Kernel kernel = SKBuilders.kernel()
            .withDefaultAIService(gpt35Turbo)
            .withDefaultAIService(gpt4)
            .withPromptTemplateEngine(new HandlebarsPromptTemplateEngine())
            .build();

        ChatHistory chatHistory = gpt35Turbo.createNewChat();
        while(true)
        {
            String input = System.console().readLine("User > ");
            chatHistory.addUserMessage(input);

            // Run the simple chat
            // The simple chat function uses the messages variable to generate the next message
            // see Plugins/ChatPlugin/SimpleChat.prompt.yaml for the full prompt
            KernelResult result = kernel.runAsync(
                true, // streaming
                ContextVariables.builder().withVariable("messages", chatHistory).build(),
                chatFunction
            ).block();

            System.console().printf("Assistant > ");
            result.functionResults().forEach(
                functionResult -> {
                    functionResult.<String>getStreamingValueAsync().subscribe(
                        message -> System.console().printf(message)
                    ); 
                    String message = functionResult.<String>getValueAsync().block();
                    chatHistory.addAssistantMessage(message);
                }
            );
        }
    }
}